websites/Q/Quizypedia/presence.ts

```diff
@@ -0,0 +1,281 @@
+const presence = new Presence({
+  clientId: '1336750164815384732',
+})
+const browsingTimestamp = Math.floor(Date.now() / 1000)
+
+const singlePageTitles: { [key: string]: string } = {
+  'about': 'À propos',
+  'accounts': 'Paramètres',
+  'blog': 'Blog',
+  'comment-creer-un-quiz': 'Aide',
+  'contact': 'Contact',
+  'edition': 'Édition',
+  'options': 'Paramètres',
+  'remarques': 'Contact',
+  'resultats': 'Classements',
+  'themes-prives': 'Thèmes privés',
+  'user': 'Paramètres',
+  'visite_guidee': 'Aide',
+}
+const themeColors: { [key: string]: string } = {
+  bgcolor1v3: 'Culture classique',
+  bgcolor2v3: 'Culture moderne',
+  bgcolor3v3: 'Culture générale',
+  bgcolor4v3: 'Géographie',
+  bgcolor5v3: 'Histoire',
+  bgcolor6v3: 'Animaux et plantes',
+  bgcolor7v3: 'Sciences et techniques',
+  bgcolor8v3: 'Sport',
+}
+
+function getUsername() {
+  if (document.querySelector('#userBtn2')) {
+    const username = document.querySelector('#userBtn2')?.innerHTML.trim()
+    return `@${username}`
+  }
+  else if (document.querySelector('#userBtn3')) {
+    return 'Non connecté'
+  }
+}
+
+presence.on('UpdateData', async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: 'https://i.imgur.com/rnJFBnZ.png',
+    startTimestamp: browsingTimestamp,
+  }
+  const [showUsername] = await Promise.all([
+    presence.getSetting<boolean>('username'),
+  ])
+  const urlData = window.location.pathname.split('/').filter(part => part)
+  const urlParams = new URLSearchParams(window.location.search)
```

Comment:
For consistency with other activities
```suggestion
  const urlData = document.location.pathname.split('/').filter(part => part)
  const urlParams = new URLSearchParams(document.location.search)
```